lms/djangoapps/learner_recommendations/utils.py

```diff
@@ -211,80 +182,23 @@ def filter_recommended_courses(
         "programs",
     ]
 
-    # Remove the course keys a user is already enrolled in
-    enrollable_course_keys = _remove_user_enrolled_course_keys(user, unfiltered_course_keys)
-
+    # Filter out enrolled courses .
+    course_ids_to_filter_out = _get_user_enrolled_course_ids(user)
     # If user is seeing the recommendations on a course about page, filter that course out of recommendations
-    recommended_course_keys = [
-        course_key
-        for course_key in enrollable_course_keys
-        if course_key != request_course
-    ]
+    if request_course_id:
+        course_ids_to_filter_out.append(request_course_id)
 
-    for course_id in recommended_course_keys:
+    for course_key in unfiltered_course_keys:
         if len(filtered_recommended_courses) >= recommendation_count:
             break
 
-        course_data = get_course_data(course_id, fields, querystring={'marketable_course_runs_only': 1})
+        course_data = get_course_data(course_key, fields, querystring={'marketable_course_runs_only': 1})
```

Comment:
Suppose a user enrolled in a course run that is not active now in a particular course. How would we filter that course from our recommendations?
Because the user has already gotten that course.
I think we should call API without `querystring`.

Reply:
we are not adding `querystring` just for filtering out enrolled courses, we also want to filter out courses with no marketable course runs. This means we are only recommending courses that are marketable.

As far as your concern, I think if a course run is not active that means its date is passed or for any reason that is not marketable which implies that the enrollment is also not active. So I guess it's fine to recommend the course if the user does not have an active enrollment in that course.